src/Farmer/Builders/Builders.Databricks.fs

```diff
@@ -0,0 +1,124 @@
+module Farmer.Builders.Databricks
+
+open Farmer
+open Farmer.Databricks
+open Farmer.Arm.Databricks
+
+type WorkspaceConfig =
+    { Name: ResourceName
+      ManagedResourceGroupId: string
+      PricingTier: PricingTier
+      DisablePublicIp: bool option
+      PrepareEncryption: bool option
+      Encryption: EncryptionConfig option
+      ByovMode: bool option
+      ByovConfig: ByovConfig option
+      Tags: Map<string,string> }
+    interface IBuilder with
+        member this.ResourceId = workspaces.resourceId this.Name
+        member this.BuildResources location = [
+            { Name = this.Name
+              Location = location
+              ManagedResourceGroupId = this.ManagedResourceGroupId
+              PricingTier = this.PricingTier
+              DisablePublicIp = this.DisablePublicIp
+              PrepareEncryption = this.PrepareEncryption
+              Encryption = this.Encryption
+              ByovMode = this.ByovMode
+              ByovConfig = this.ByovConfig
+              Tags = this.Tags  }
+        ]        
+
+type WorkspaceBuilder() =
+    member __.Yield _ =
+      { 
+        ///Name of the workspace
+        Name = ResourceName ""
+        ///Azure managed resource group name
+        ManagedResourceGroupId = ""
+        ///Workspace pricing tier
+        PricingTier = StandardTier
+        ///Should public IPs be used
+        DisablePublicIp = None
+        ///Prepare workspace for encryption
+        PrepareEncryption = None
+        ///Encryption configuration
+        Encryption = None
+        ///Should use Bring Your Own VNet networking model
+        ByovMode = None
+        ///Bring Your Own VNet configuration
+        ByovConfig = None
+        ///List of tags
+        Tags = Map.empty
+      }
+    ///Set workspace name  
+    [<CustomOperation "name">]
+    member __.Name (state:WorkspaceConfig, name) =
+        { state with Name = ResourceName name }
+    ///Set managed resource group
+    [<CustomOperation "managed_resource_group_id">]
+    member _.ManagedResourceGroupId (state:WorkspaceConfig, resourceGroupName) =
+        { state with ManagedResourceGroupId = resourceGroupName}
+    ///Set workspace pricing tier
+    [<CustomOperation "pricing_tier">]
+    member _.PricingTier (state:WorkspaceConfig, sku) =
+        { state with PricingTier = sku }
+    ///Set public IP enablement
+    [<CustomOperation "disable_public_ip">]
+    member _.DisablePublicIp(state:WorkspaceConfig) =
+        { state with DisablePublicIp = Some true}
+    ///Set encryption preparation
+    [<CustomOperation "enable_encryption">]
+    member _.PrepareEncryption(state:WorkspaceConfig) =
+        { state with 
+            PrepareEncryption = Some true
+            Encryption = Some {
+                KeySource = "Microsoft.Keyvault";
+                KeyName = "";
+                KeyVersion = "latest";
+                KeyVaultUri = ""} }
+    ///Set encryption key name
+    [<CustomOperation "key_name">]
```

Comment:
Might it be worth merging some of these keywords together - it looks like you should never set the Key Vault without setting the name / version. I also wonder whether we can just omit the `enable_encryption` keyword - if you supply a keyvault, it enables encryption implicitly?

Reply:
This has been changed to an operation called "encryption_key" that takes in a name and version. The version also defaults to "latest".